src/submit.rs

```diff
@@ -476,4 +477,44 @@ impl<'a> Submitter<'a> {
         )
         .map(drop)
     }
+
+    /// Performs a synchronous cancellation request, similiar to [AsyncCancel](crate::opcode::AsyncCancel),
+    /// except that it completes synchronously.
+    ///
+    /// [`AsyncCancelFlags`](types::AsyncCancelFlags) may be supplied to indicate the strategy for
+    /// matching operations to cancel.
+    ///
+    /// Available since 6.0.
```

Comment:
This comment block should probably indicate what values to use for the optional values. Like 0 for user_data and -1 for fd. The -1 for fd wasn't obvious to me until I saw the test case use it. Passing zero when a library like tokio-uring uses slab indexes where zero is very common would make me nervous but a little testing with the correct flag set would resolve that it was okay for me. I might just pass the max value allowed anyway making it more obvious.

It's a little strange to me that only the timeout parameter actually would take an Option<T> but that doesn't bother quininer so I am in the minority, so just a comment, not a request for change. Other possibilities were all three are options, or the first two are a single enum with two invariants. But I understand this library's job is to provide bindings, not necessarily the strictest API surface for the promised semantics.

Reply:
Thanks! I agree the arguments are pretty confusing at the moment so any way we can make them safer or more obvious sounds good to me. 

> The -1 for fd wasn't obvious to me
I believe you can pass any value in for the FD as long as you do not set the `AsyncCancelFlag::FD/FD_FIXED` flag. I was just passing in `-1` to try and make it obvious that we were not canceling by the FD we were reading from in the tests. 

Maybe we could encode the match criteria in a builder-like thing. 

For example:
```
types::MatchOn::fd(types::Fd(4)).and_user_data(42).match_any();
```
This would match any requests where either the FD = 42 or the user_data = 42.

With this, the signature would be
```
pub fn register_sync_cancel(&self, match_on: types::MatchOn, timeout: Option<Timespec>) -> io::Result<()>;
```
This might be the way to go because we can't really support `Option<impl UseFixed>` directly in the API, because the compiler will complain about the concrete type being unknown if we supply `None` for the FD. 

I'll try this out and we can see if it's worth adding another builder type. c